### ISSUE TITLE
send_video lost file extension

### ISSUE THREAD

**Issue by pista86** [vysheng/tg#401]
When I run the following command:

> send_video Marcelo_Pistarelli /bunny.mp4
36 [14:52]  Marcelo Pistarelli <<< [video: type=video/mp4 size=100x100 duration=100 size=5MiB]

I get on my android phone but without the file extension mp4, then the telegram application recognizes it as a generic file without extension and not allow playback from the application.
Telegram: +54346909209

**Comment by malishev** [comment]
I have the same issue using the `send_document()` function.

The file looks like this (no extension):
![2015-01-28 17 20 35](https://cloud.githubusercontent.com/assets/2178368/5939106/043cdb76-a712-11e4-915d-52f0efd71cfe.png)

Log entry is like that:
```
[17:17]  Admin <<< [document: type=application/octet-stream size=28KiB]
```

And the error is like this (webogram console):
```
Error: t.file_name is undefined
```

**Comment by kalehmann** [comment]
Same problem with send_audio.
On mobile clients, the file is unusual without the correct extension :-(

`> send_audio peer file.ogg`
`[15:36]  peer <<< [audio: type=audio/ogg duration=100 size=9KiB]`

**Comment by eliafino** [comment]
Me too for sending video...
... please, fix ^_^

**Comment by kalehmann** [comment]
I've found a workaround: using an older version of tg.
Now I use https://github.com/vysheng/tg/archive/1.0.4.zip and sending audio, photo and video works fine.

**Comment by eliafino** [comment]
Thanks kalemann,
but in 1.0.4 not present "port to listen for input commands"

**Comment by eliafino** [comment]
please fix :-(

**Comment by jonnywilliamson** [comment]
Yes the same thing is happening if you send a jpg image with send_document.

The recipient gets a thumbnail of the jpg, but when they try and open it, it shows "file" with no file extension and you can't see the image.

Sending the same image via send_photo works fine.

Also, a gif file appears to work, but not a jpg. What a shame.

**Comment by eliafino** [comment]
same for video with send_video
if you send a video, extension is removed and no way to open...

2015-02-24 15:29 GMT+01:00 Jonathan <notifications@github.com>:

**Comment by terany** [comment]
One more vote for the fix

**Comment by jonnywilliamson** [comment]
If you are interested in a work around, you can use a slightly older version of the code.

Checkout this commit for the tg repository

https://github.com/vysheng/tg/commit/4630862924102ce508132296670c7c712bc65ee2

And this commit in the tgl repository:
https://github.com/vysheng/tgl/commit/0205b668d3abcf35a348556776d1180439d115d9

do a ./configure and a make and it's back working again.

**Comment by MartinzSmitt** [comment]
I have too issue using the send_document() function.
![img](https://cloud.githubusercontent.com/assets/11694955/6881297/0f117e68-d56a-11e4-9c82-bcd6553547ce.png)
Please, fix it
